src/condor_utils/checksum.h

```diff
@@ -0,0 +1,28 @@
+/***************************************************************
+ *
+ * Copyright (C) 2022, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+#ifndef _COMPUTE_CHECKSUM_H
+#define _COMPUTE_CHECKSUM_H
+
+// #include <string>
+
+bool compute_checksum( int fd, std::string & checksum );
+bool compute_file_checksum( const std::string & file_name, std::string & checksum );
```

Comment:
These function names should mention what checksum algorithm is being used.

Reply:
Should they?  That's only going to cause us to have make a bunch of code changes when we change the algorithm later, right?

Reply:
If we change the algorithm later, we need to be acutely aware of the change for backwards compatibility. Doubly so since these functions are not tied to the manifest code.